### PR TITLE
Add types to permissions.tsx

### PR DESCRIPTION
**Pull request by tomasmatus**

### CODE REVIEW

File: src/dialogs/permissions.tsx [hunk 1]
```diff
@@ -110,23 +117,29 @@ function mode_to_args(mode) {
...
     if (!selected) {
         const directory_name = basename(path);
-        selected = { ...cwdInfo, isCwd: true, name: directory_name };
+        selected = { ...cwdInfo, name: directory_name, category: null, to: null };
+        selectedIsCwd = true;
     }
 
     const [owner, setOwner] = useState(selected.user);
-    const [mode, setMode] = useState(selected.mode);
+    const [mode, setMode] = useState(selected.mode ?? 0);
```
Comment (cockpituous) [inline src/dialogs/permissions.tsx:136]: This added line is not executed by any test.

File: src/dialogs/permissions.tsx [hunk 2]
```diff
@@ -146,32 +159,35 @@ const EditPermissionsModal = ({ dialogResult, selected, path }) => {
...
+            const full_path = selectedIsCwd ? path : path + selected.name;
+            const selinux_context = await python.spawn(read_selinux_context, [full_path]);
             setSELinuxContext(selinux_context);
-        } catch (exc) {
-            if (exc.exit_status !== 2)
-                console.error("Cannot obtain SELinux context", exc);
+        } catch (err) {
+            const e = err as python.PythonExitStatus;
+            if (e.exit_status !== 2)
+                console.error("Cannot obtain SELinux context", err);
```
Comment (cockpituous) [inline src/dialogs/permissions.tsx:168]: These 4 added lines are not executed by any test.

File: src/dialogs/permissions.tsx [hunk 3]
```diff
@@ -258,9 +275,9 @@ const EditPermissionsModal = ({ dialogResult, selected, path }) => {
           variant={ModalVariant.small}
           /* Translators: $0 represents a filename */
           title={fmt_to_fragments(_("$0 permissions"), <b>{selected.name}</b>)}
-          description={inode_types[selected.type] || "Unknown type"}
+          description={(selected.type) ? inode_types[selected.type] : "Unknown type"}
```
Comment (cockpituous) [inline src/dialogs/permissions.tsx:278]: This added line is not executed by any test.

**Comment by jelly** [comment]
Green on first try?! :partying_face:

File: src/dialogs/permissions.tsx [hunk 4]
```diff
@@ -258,9 +275,9 @@ const EditPermissionsModal = ({ dialogResult, selected, path }) => {
...
           isOpen
-          onClose={dialogResult.resolve}
+          onClose={() => dialogResult.resolve()}
```
Comment (jelly) [inline src/dialogs/permissions.tsx:280]: Hmm why is this not happy for typing?

File: src/dialogs/permissions.tsx (hunk 3 above)
Comment (jelly) [inline src/dialogs/permissions.tsx:278]: This one I don't understand? Also easy test coverage fix :D And hmm even worse missing translations this should be:

`_("Missing type")` woops!

File: src/dialogs/permissions.tsx [hunk 5]
```diff
@@ -110,23 +117,29 @@ function mode_to_args(mode) {
...
     const [group, setGroup] = useState(selected.group);
-    const [errorMessage, setErrorMessage] = useState(undefined);
-    const [accounts, setAccounts] = useState(null);
-    const [groups, setGroups] = useState(null);
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
+    const [accounts, setAccounts] = useState<PasswdUserInfo[] | null>(null);
+    const [groups, setGroups] = useState<EtcGroupInfo[] | null>(null);
     const [isExecutable, setIsExecutable] = useState((mode & 0b001001001) === 0b001001001);
-    const [selinuxContext, setSELinuxContext] = useState(null);
+    const [selinuxContext, setSELinuxContext] = useState("");
```
Comment (jelly) [inline src/dialogs/permissions.tsx:142]: This seems wrong, we compare `!== null` when rendering? so this should be the same as above `string | null`

File: src/dialogs/permissions.tsx (hunk 4 above)
Comment (tomasmatus) [reply src/dialogs/permissions.tsx:280]: `Type '(value: void) => void' is not assignable to type '(event: KeyboardEvent | MouseEvent<Element, MouseEvent>) => void'.

File: src/dialogs/permissions.tsx (hunk 3 above)
Comment (tomasmatus) [reply src/dialogs/permissions.tsx:278]: it doesn't really like that `selected.type` can be `undefined` when directly using it as index type. And yeah, translation! 😮

File: src/dialogs/permissions.tsx (hunk 5 above)
Comment (tomasmatus) [reply src/dialogs/permissions.tsx:142]: oh yeah, my bad

File: src/dialogs/permissions.tsx (hunk 1 above)
Comment (cockpituous) [inline src/dialogs/permissions.tsx:136]: This added line is not executed by any test.

File: src/dialogs/permissions.tsx (hunk 2 above)
Comment (cockpituous) [inline src/dialogs/permissions.tsx:168]: These 4 added lines are not executed by any test.

File: src/dialogs/permissions.tsx [hunk 6]
```diff
@@ -258,9 +275,9 @@ const EditPermissionsModal = ({ dialogResult, selected, path }) => {
           variant={ModalVariant.small}
           /* Translators: $0 represents a filename */
           title={fmt_to_fragments(_("$0 permissions"), <b>{selected.name}</b>)}
-          description={inode_types[selected.type] || "Unknown type"}
+          description={(selected.type) ? inode_types[selected.type] : _("Missing type")}
```
Comment (cockpituous) [inline src/dialogs/permissions.tsx:278]: This added line is not executed by any test.

**Review by jelly — APPROVED** [review]
Cool stuff

**Pull request merged**